flake-modules/lynx-docs/default.nix

```diff
@@ -0,0 +1,123 @@
+_: { self, inputs, config, lib, pkgs, ... }:
+{
+  config.perSystem = args@{ config, self', inputs', pkgs, lib, system, ... }:
+    let generate = pkgs.callPackage ({
+        lib
+        , name
+        , runCommand
+        , nixosOptionsDoc
+        , gnused
+        , coreutils
+        , modules ? []
+        , htmlSafe ? false
+        , uri ? null
+        , ...
+      }:
+        let
+          eval = lib.evalModules {
+            # maybe scary and wrong.
+            # but emperically, it works.
+            modules = map (x: x // { config._module.check = false; }) modules;
+          };
+
+          docs = nixosOptionsDoc {
+            inherit (eval) options;
+            warningsAreErrors = false;
+          };
+        in
+          runCommand name {
+            buildInputs = [ gnused coreutils ];
+          }
+          ''
+            cat ${docs.optionsCommonMark} \
+              ${if htmlSafe then
+                ''| sed -e "s|\\\<|\&lt\;|g"'' else "" } \
+              ${if uri != null then
+                ''| sed -e "s|file:///nix/store/.*-source/|${uri}|g"''
+                else ""} \
+              > $out
+          ''
+      );
```

Comment:
I'll rework this hack into flake-module options.

Reply:
Also worth noting that breaking apart the htmlSafe & relinking phases into a separate drv would be ideal